data_integration/notification/notifier.py

```diff
@@ -0,0 +1,75 @@
+"""Notifies the ChatRoom to send message to channels"""
+
+from data_integration.notification.chat_room import ChatRoom
+from data_integration.logging import pipeline_events
+from data_integration import events
+
+
+class Notifier(events.EventHandler):
+
+    def __init__(self, chat_rooms: [ChatRoom]):
+        self.chat_rooms = chat_rooms
+        self.node_output: {tuple: {bool: [events.Event]}} = None
+
+    def handle_event(self, event: events.Event):
+        """
+        Send the output of a node when event occurs.
+        Args:
+            event: The current event of interest
+        """
+
+        if isinstance(event, pipeline_events.Output):
+            key = tuple(event.node_path)
+
+            if not self.node_output:
+                self.node_output = {}
+
+            if not key in self.node_output:
+                self.node_output[key] = {True: [], False: []}
+
+            self.node_output[key][event.is_error].append(event)
+
+        elif isinstance(event, pipeline_events.NodeFinished):
+            key = tuple(event.node_path)
+            if not event.succeeded and event.is_pipeline is False:
+                for chat_room in self.chat_rooms:
+
+                    text = chat_room.create_error_text(node_path=event.node_path)
+
+                    log = ''

```

Comment:
log = None
error_log =None

Feels cleaner for me.

Reply:
Yes, you are right